src/checksum_verify.ts

```diff
@@ -0,0 +1,20 @@
+import * as core from "@actions/core";
+import * as crypto from "crypto"
+import * as fs from "fs"
+
+export function checksumVerify(downloadPath: string){
+
+
+    const fileBuffer:Buffer = fs.readFileSync(downloadPath)
+    const checksum: string = crypto.createHash("sha256").update(fileBuffer).digest('hex'); // checksum of downloaded file
+
+    const expectedChecksum: string = core.getInput("expected_checksum") // default checksum
+    
+    if(checksum !== expectedChecksum){
+        core.error(`Checksum verification failed.`)
```

Comment:
Do we need the core.error separately? Or can the line `Checksum verification failed` be merged with the core.setFailed one.

Reply:
@varunsh-coder merged `Checksum verification failed` in `core.setFailed`